lfs/upload_queue.go

```diff
@@ -100,7 +101,9 @@ func ensureFile(smudgePath, cleanPath string) error {
 		return err
 	}
 
-	cleaned.Close()
+	if cleaned != nil {
+		defer cleaned.Teardown()
```

Comment:
I don't think this section is quite right, for a couple reasons. Primarily, it's possible for `PointerClean` to return an error with and a valid `cleanedAsset` if the [CopyWithCallback](https://github.com/github/git-lfs/blob/8b3fa6aec1414a00f609f900b934b09d302ef5a1/lfs/pointer_clean.go#L46) returns an error. In this case, `ensureFile` will return without removing the cleaned file.

Less importantly, I don't think we need to defer the `Teardown()`, It's only removing the underlying file and we're not using it before the end of the function.

Maybe something like:
```diff
diff --git a/lfs/upload_queue.go b/lfs/upload_queue.go
index cbd075a..4a26f08 100644
--- a/lfs/upload_queue.go
+++ b/lfs/upload_queue.go
@@ -97,12 +97,12 @@ func ensureFile(smudgePath, cleanPath string) error {
 	}
 
 	cleaned, err := PointerClean(file, stat.Size(), nil)
-	if err != nil {
-		return err
+	if cleaned != nil {
+		cleaned.Teardown()
 	}
 
-	if cleaned != nil {
-		defer cleaned.Teardown()
+	if err != nil {
+		return err
 	}
 
 	if expectedOid != cleaned.Oid {
```